pkg/graveler/committed/apply.go

```diff
@@ -26,99 +26,129 @@ func incrementDiffSummary(d *graveler.DiffSummary, typ graveler.DiffType) {
 
 // ReferenceType represents the type of the reference
 
-// applyFromSource applies all changes from source to writer.
-func applyFromSource(ctx context.Context, logger logging.Logger, writer MetaRangeWriter, source Iterator) error {
+// applyFromIter applies all changes from Iterator to writer.
+// returns the number of writes
+func applyFromIter(ctx context.Context, logger logging.Logger, writer MetaRangeWriter, iter Iterator) (int, error) {
+	var count int
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
+			return 0, ctx.Err()
 		default:
 		}
-
-		sourceValue, sourceRange := source.Value()
-		if sourceValue == nil {
-			if logger.IsTracing() {
-				logger.WithFields(logging.Fields{
-					"from": string(sourceRange.MinKey),
-					"to":   string(sourceRange.MaxKey),
-					"ID":   sourceRange.ID,
-				}).Trace("copy entire source range at end")
-			}
-			if err := writer.WriteRange(*sourceRange); err != nil {
-				return fmt.Errorf("copy source range %s: %w", sourceRange.ID, err)
+		iterValue, iterRange := iter.Value()
+		if iterValue == nil {
+			if iterRange.IsTombstone() {
+				// internal error but no data lost: deletion requested of a
```

Comment:
This is not counted, I assume on purpose because it's a probably bug?  Maybe still count unjustified deletes just to be sure...

Reply:
`applyFromIter` only returns addition because it runs at when there is no iterator to compare with, so it should have only additions. I could add "deletions" to it, but I don't think It should be counted. @arielshaqed  what do you think ?

Reply:
Dunno.  I'd count it as a separate category maybe, that way if it starts showing up we have a chance of someone complaining.